sdk/core/System.ClientModel/src/Internal/SSE/ServerSentEvent.cs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace System.ClientModel.Internal;
+
+internal struct PendingEvent
+{
+    private List<ServerSentEventField>? _dataFields;
+
+    public int DataLength { get; set; }
+    public List<ServerSentEventField> DataFields => _dataFields ??= new();
+    public ServerSentEventField? EventNameField { get; set; }
+    public ServerSentEventField? IdField { get; set; }
+    public ServerSentEventField? RetryField { get; set; }
+}
+
+// SSE specification: https://html.spec.whatwg.org/multipage/server-sent-events.html#parsing-an-event-stream
+internal readonly struct ServerSentEvent
+{
+    private const char LF = '\n';
+
+    // Gets the value of the SSE "event type" buffer, used to distinguish between event kinds.
+    public ReadOnlyMemory<char> EventName { get; }
+    // Gets the value of the SSE "data" buffer, which holds the payload of the server-sent event.
+    public ReadOnlyMemory<char> Data { get; }
+    // Gets the value of the "last event ID" buffer, with which a user agent can reestablish a session.
+    public ReadOnlyMemory<char> LastEventId { get; }
+    // If present, gets the defined "retry" value for the event, which represents the delay before reconnecting.
+    public TimeSpan? ReconnectionTime { get; }
```

Comment:
Similar comment to https://github.com/Azure/azure-sdk-for-net/pull/43840/files#r1594450028 about resuming lost connections.

Reply:
https://github.com/Azure/azure-sdk-for-net/pull/43840#discussion_r1594840669